### ISSUE TITLE
Automatically mounting volumes, configmaps, and secrets

### ISSUE THREAD

**Issue by kadel** [redhat-developer/odo#6549]
- [x] Implement specifications at https://github.com/devfile/devworkspace-operator/blob/main/docs/additional-configuration.adoc#automatically-mounting-volumes-configmaps-and-secrets
- [x] Document behaviour (or point to the specification)
- [x] integration tests
- [x] Check with @kadel / @l0rd the new label replacing `controller.devfile.io/mount-to-devworkspace`
- [x] Only on cluster
- [ ] Implement controller.devfile.io/mount-access-mode (#6728)

related to #6478 epic

**Comment by valaparthvi** [comment]
@kadel Can you set a priority on this?

**Comment by valaparthvi** [comment]
Related to #6542.

**Comment by kadel** [comment]
Bumping priority as we have a user requiring this to use odo in their environment.

**Comment by feloy** [comment]
@kadel, What is the purpose of `controller.devfile.io/watch-configmap` and `controller.devfile.io/watch-secret` labels? (https://github.com/devfile/devworkspace-operator/blob/main/docs/additional-configuration.adoc#automatically-mounting-volumes-configmaps-and-secrets)

**Comment by kadel** [comment]
> @kadel, What is the purpose of `controller.devfile.io/watch-configmap` and `controller.devfile.io/watch-secret` labels? (https://github.com/devfile/devworkspace-operator/blob/main/docs/additional-configuration.adoc#automatically-mounting-volumes-configmaps-and-secrets)

not sure :-( 




> @kadel Do we want to implement this for Podman?

Not for now. If I'm not mistaken, Secrets and ConfigMaps are not supported on Podman. Once podman adds support for those we should implement this. 

Volumes are supported to some extent on Podman, but I don't think it makes sense to do just partial implementation for volumes. We will add all this together with Secrets and ConfigMaps

**Comment by feloy** [comment]
@kadel What would be the new label replacing `controller.devfile.io/mount-to-devworkspace`?

**Comment by feloy** [comment]
@l0rd Do you have an opinion on what should be the new label replacing `controller.devfile.io/mount-to-devworkspace` to have a label used by all tools?

**Comment by l0rd** [comment]
@feloy I was thinking about `controller.devfile.io/auto-mount` but I also like the one that you have in the PR `controller.devfile.io/mount-to-containers`. @amisevsk @AObuchow what do you think?

**Comment by amisevsk** [comment]
> @kadel, What is the purpose of controller.devfile.io/watch-configmap and controller.devfile.io/watch-secret labels? (https://github.com/devfile/devworkspace-operator/blob/main/docs/additional-configuration.adoc#automatically-mounting-volumes-configmaps-and-secrets)

These labels are required for the DevWorkspace Operator (and should not be necessary for `odo`). In operator land, we don't want to cache every configmap/secret on the cluster internally, so we filter by label. Since configmaps/labels aren't associated with any one workspace, we need a more generic label for "relevant to DWO".

> I was thinking about controller.devfile.io/auto-mount but I also like the one that you have in the PR controller.devfile.io/mount-to-containers.

If we're moving the label definition up to devfile/api, we should modify the `controller.devfile.io` part -- that is included to scope labels/annotations to the DevWorkspace Operator (as opposed to the general devfile API). As for specific naming, I like `auto-mount` as it's shorter, but worry it may be just the language _we_ use for the feature and so confusing to users.

**Comment by kadel** [comment]
I was the one who originally suggested keeping `controller.devfile.io` to Mario, I thought that it will make things easier for DWO, as most of the labels could stay as they are. 

`controller.devfile.io` is generic enough that I think we could keep it.
I'm more than fine with using something like `devfile.io/...` if DWO supports that as well.

**Comment by amisevsk** [comment]
If we're content with adopting `controller.devfile.io`, then that works too -- it's up to the collective conscious :). It certainly would be easier for DWO, at least.

My main concern would be long term -- we should decide what prefix/name we want to use for devfile API-defined labels and annotations and stick to that going forward (e.g. `controller.devfile.io`, `devfile.io`, etc.). When we started requiring specific labels that the DevWorkspace Operator adheres to, we opted for `controller.devfile.io` -- this is used for other DWO-specific labels, e.g. `controller.devfile.io/restricted-access`, `controller.devfile.io/storage-type`, etc.
